provider/runbook_data_test.go

```diff
@@ -0,0 +1,102 @@
+package provider
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccRunbookDataSource_basic(t *testing.T) {
+	rName := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRunbookDataSourceConfig_basic(rName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.firehydrant_runbook.test_runbook", "id"),
+					resource.TestCheckResourceAttr(
+						"data.firehydrant_runbook.test_runbook", "name", fmt.Sprintf("test-runbook-%s", rName)),
```

Comment:
Just to clarify this is testing that when the resource comes back that the `rName` just a random CharSet that you created above gets returned as well as that there is a `id` attached?

Reply:
Yeah! This is leaning pretty heavily on the Terraform provider acceptance test framework, which you can [read more about here](https://www.terraform.io/plugin/sdkv2/testing/acceptance-tests).

This is essentially testing that, after applying the Terraform config returned by`testAccRunbookDataSourceConfig_basic(rName)`
1. the `id` is set in state for the data source `data.firehydrant_runbook.test_runbook`
1. the `name` is set in state and matches the value we expect (which comes from the Terraform config we used here) for the data source `data.firehydrant_runbook.test_runbook`